### ISSUE TITLE
Namespaced directory names?

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/env-paths#1]
For example `envPaths('unicorn').config` outputs `/Users/sindresorhus/Library/Application Support/unicorn` on macOS. Imagine if a desktop app is named `unicorn` too. There would be conflict as desktop apps have config there too. Not very likely, but not sure I'm willing to take the risk either.

Do you think they should be namespaced? For example: `/Users/sindresorhus/Library/Application Support/nodejs-unicorn`? That would remove all possible conflict, although more ugly.

-

tl;dr `unicorn` → `nodejs-unicorn`

**Comment by sindresorhus** [comment]
// @jamestalmage @SamVerschueren @kevva @paulmolluzzo @vdemedes

**Comment by paulmolluzzo** [comment]
I don't know that namespacing is a guaranteed resolution. Wouldn't you make the app name unique to begin with anyway?

If you're trying to provide a method for setting a location for saving data locally the OS has its own name conflict resolution process, e.g. having two apps named `unicorn` would make the second one have a config like `/Users/sindresorhus/Library/Application Support/unicorn\ 2`. However it would be tough to know which of the two (or more) similarly named directories you'd use. Assuming a perfectly unique name it almost wouldn't matter - what if you install the app again? And even that is kind of irrelevant: If you install two instances of `unicorn` the config would be identical I think, no?

Cocoa seems to have a similar function available that just returns the Application Support (or Library or whatever) directory, leaving the naming process, conflict resolution, and subsequent "finding" process nearly the same as far as I can tell:

http://www.cocoawithlove.com/2010/05/finding-or-creating-application-support.html

http://stackoverflow.com/questions/11804309/where-can-a-sandboxed-mac-app-save-files

**Comment by jamestalmage** [comment]
I say just make this change now, before people really start using it. The risk is definitely small, but the cost to the user to avoid that risk is almost zero. Almost no one will ever look at these directories, and if they do it will only be once or twice to debug something. The "ugliness" of providing a robust solution isn't going to affect their daily lives at all (and a collision could potentially cause major headaches).

I might suffix it instead (`unicorn-nodejs` instead of `nodejs-unicorn`), that puts it in a better order when `ls`-ing the directory contents, and allows you to tab to autocomplete the `-nodejs` part in the terminal.

**Comment by sindresorhus** [comment]
> Wouldn't you make the app name unique to begin with anyway?

You don't control what other people name their apps. Even in different ecosystems. An app could have the name `unicorn` in both Node.js and macOS.

**Comment by sindresorhus** [comment]
> I might suffix it instead (unicorn-nodejs instead of nodejs-unicorn), that puts it in a better order when ls-ing the directory contents, and allows you to tab to autocomplete the -nodejs part in the terminal.

👍

**Comment by sindresorhus** [comment]
Thanks for the feedback guys :)
